File: coffee-time-challenges/06-word-doc/main.py
#!/usr/bin/env python

number_list = [str(i) for i in range(10000+1)]
daughter_list = map(lambda n: n.replace('0', ' '), number_list)
numbers = []
for el in daughter_list:
    for n in el.split(' '):
        if len(n) > 0:
            numbers.append(int(n))
print(sum(numbers))

File: coffee-time-challenges/06-word-doc/main.scala
object WordDoc {
    def main(arg: Array[String]) {
        val number_list = for(i <- 1 to 10000) yield i.toString;
        val daughter_list = for(el<-number_list) yield el.replace('0', ' ');
        var sum = 0;
        for(el <- daughter_list) {
            for(n <- el.split(' ')) {
                if(n.length > 0){
                    sum += n.toInt;
                }
            }
        }
        println(sum);
    }
}